python/cog/server/http.py

```diff
@@ -79,9 +81,20 @@ def create_app(
         input_type=InputType, output_type=OutputType
     )
 
+    http_semaphore = asyncio.Semaphore(threads)
+
+    def limited(f: Callable) -> Callable:
+        @functools.wraps(f)
+        async def wrapped(*args: Any, **kwargs: Any) -> Any:
+            async with http_semaphore:
+                return await f(*args, **kwargs)
+
+        return wrapped
+
     @app.on_event("startup")
     def startup() -> None:
         # https://github.com/tiangolo/fastapi/issues/4221
+        # potentially no longer necessary?
         RunVar("_default_thread_limiter").set(CapacityLimiter(threads))  # type: ignore
```

Comment:
I think it's definitely no longer necessary if we've verified that the semaphore does the same job.